### DISCUSSION TITLE
Store Multi-Select Picker results in a bbdb table?

### DISCUSSION THREAD

**Discussion by gerhardmpl** [Budibase/budibase#4741] [Help]
I have a table `options` that I want to use as a Data Provider for a `multi-select picker`:

`options` (id: number 0-99, name: text, name_display: formular, {{id}} - {{name}}

where _name_display_ is the `Label column` and _id_ is the `Value column` for the field _selectedOptions_. I want to store and later use the result in another ddbd table and can not figure out, which data type to use. 

Output from `return $("Form.Fields.selectedOptions");` is something like `1,3,9` and `typeof` tells me, it is an `object`. None of the available data types works, so I doubt there is an adaquate data type available to store the mulit-select picker results in a bbdb table? Or do I need to transform the object?

**Comment by mjashanks** [comment]
Hi @gerhardmpl 

I got most of the way there with this:
- I can save the values ok
- I haven't worked out how to load the values into the multi-picker i.e. on an edit form.

So, 
1. I created a column called `tags` which is a `JSON` type. The schema for the JSON type is as follows

```json
{
  "tags" : [ 1 , 2, 3 ]
}
```
i.e. it has a field called "tags", which is an array of numbers (IDs in my options table)

2. Created my options table as detailed by you
3. Added my `Data Provider` + `Multi Picker` as above
4. I set the `Field` of my picker to `tags_picker`
5. In the `Save Row` action, I override the `Tags` column, and set the value with the following javascript

```javascript
return {
	tags: $("Form.Fields.Tags_picker")
};
```
**this all works**

The issue I ran up against, was when trying to get the multi picker to load already saved values. So, I set the `Default Value` field on the picker to the following javascript:

```javascript
return $("Form.Fields.tags.tags") || [] ;
```

This correctly loads the values in, however, it won't let me remove values from the list - I'm not quite sure what's going on. Maybe you could play with that!
